core/interfaces/i_drag_target.js

```diff
@@ -58,3 +58,9 @@ Blockly.IDragTarget.prototype.onBlockDrop;
  * @param {!Blockly.IBubble} bubble The bubble.
  */
 Blockly.IDragTarget.prototype.onBubbleDrop;
+
+/**
+ * Whether a block move should be prevented if it is dropped on this component.
+ * @type {boolean}
+ */
+Blockly.IDragTarget.prototype.preventBlockMove;
```

Comment:
Some variations to consider:
- passing in block as parameter and make a method
- creating a parallel property/method for bubbles